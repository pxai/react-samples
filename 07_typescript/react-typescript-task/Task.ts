class Task {
    private _id: number;
    private _name: string;

    public constructor (name: string = "") {
        this._id = Math.round(1000 * Math.random());
        this._name = name;
    }

    public get id (): number {
        return this._id;
    }

    public get name (): string {
        return this._name;
    }

    public set name (name: string) {
        this._name = name;
    }
}

export default Task;