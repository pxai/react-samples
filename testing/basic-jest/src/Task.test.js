import React from 'react';
import ReactDOM from 'react-dom';
import Task from './Task';
import renderer from 'react-test-renderer';

test('renders without crashing', () => {
  const div = document.createElement('div');
  const props = { task : { id: 1, task: "test" } };
  ReactDOM.render(<Task task={props.task} />, div);
  ReactDOM.unmountComponentAtNode(div);
});


test('it matches snapshot', () => {
	const component = renderer.create(<Task />);
 let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

