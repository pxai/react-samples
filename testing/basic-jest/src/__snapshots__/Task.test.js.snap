// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<div>
  <div>
    1
    - 
    test
  </div>
  <div>
    <a
      href="javascript:void(0)"
      onClick={[Function]}
    >
      Update
    </a>
     |
    <a
      href="javascript:void(0)"
      onClick={[Function]}
    >
      Delete
    </a>
  </div>
</div>
`;
